tests/nc-client/main.c

```diff
@@ -64,9 +64,13 @@ callback_wrote(void * cookie, ssize_t lenwrit)
 	/* We are no longer writing. */
 	send->write_cookie = NULL;
 
-	/* Check that we wrote all our data. */
-	if (lenwrit != send->nchars) {
-		warn0("Mismatch between data sent and data requested to send");
+	/* Check for failure cases. */
+	if (lenwrit == -1) {
+		warnp("network_write send");
+		goto err0;
+	} else if (lenwrit != send->nchars) {
+		warnp("Mismatch between data sent: %zu, and data "
```

Comment:
It doesn't matter all that much in this case, but this really should be an assert, since the valid values passed to the network_write callback are specified by the API.  Warnings are for "something external failed" (we ran out of memory, there was a network error, etc.), but we use assertions for "we have observed a bug in our code" since that triggers a core dump (which is useful for figuring out why our code is broken).